tests/python_package_test/test_basic.py

```diff
@@ -997,3 +997,39 @@ def test_equal_datasets_from_one_and_several_matrices_w_different_layouts(rng, t
     lgb.Dataset(mat)._dump_text(one_path)
 
     assert filecmp.cmp(one_path, several_path)
+
+
+def test_string_serialized_params_retrieval(rng):
+    # Random train data
+    train_x = rng.random((500, 20))
+    train_y = rng.integers(0, 1, 500)
+    train_data = lgb.Dataset(train_x, train_y)
+
+    # Parameters
+    params = {
+        "boosting": "gbdt",
```

Comment:
```suggestion
        "boosting": "gbdt",
        "deterministic": true,
        "feature_contri": [0.5] * train_x.shape[1],
        "interaction_constraints": [[0, 1], [0]],
```

This test is currently checking that the following types of parameters successfully make it through this serialization-deserialization round trip:

* string
* int
* float
* array of string

Let's please also test other types:

* boolean
* array of float
* array of arrays of integers